src/Microsoft.ML.StandardTrainers/Standard/SdcaMulticlass.cs

```diff
@@ -412,34 +413,131 @@ private protected override SchemaShape.Column[] GetOutputColumnsCore(SchemaShape
+    {
+        public class Options : CommonOptions
+        {
+        }
+
+        internal SdcaMulticlassClassificationTrainer(IHostEnvironment env,
+            string labelColumn = DefaultColumnNames.Label,
+            string featureColumn = DefaultColumnNames.Features,
+            string weights = null,
+            float? l2Const = null,
+            float? l1Threshold = null,
+            int? maxIterations = null)
+             : base(env, labelColumn: labelColumn, featureColumn: featureColumn, weights: weights, loss: new LogLoss(),
+                   l2Const: l2Const, l1Threshold: l1Threshold, maxIterations: maxIterations)
+        {
+        }
+
+        internal SdcaMulticlassClassificationTrainer(IHostEnvironment env, Options options,
+            string featureColumn, string labelColumn, string weightColumn = null)
+            : base(env, options: options, featureColumn: featureColumn, labelColumn: labelColumn, weightColumn: weightColumn)
+        {
+        }
+
+        internal SdcaMulticlassClassificationTrainer(IHostEnvironment env, Options options)
+            : base(env, options)
+        {
+        }
+
+        private protected override MaximumEntropyModelParameters CreatePredictor(VBuffer<float>[] weights, float[] bias)
         {
             Host.CheckValue(weights, nameof(weights));
             Host.CheckValue(bias, nameof(bias));
             Host.CheckParam(weights.Length > 0, nameof(weights));
             Host.CheckParam(weights.Length == bias.Length, nameof(weights));
 
-            return new MulticlassLogisticRegressionModelParameters(Host, weights, bias, bias.Length, weights[0].Length, null, stats: null);
+            return new MaximumEntropyModelParameters(Host, weights, bias, bias.Length, weights[0].Length, null, stats: null);
         }
 
-        private protected override void CheckLabel(RoleMappedData examples, out int weightSetCount)
+        private protected override MulticlassPredictionTransformer<MaximumEntropyModelParameters> MakeTransformer(
+            MaximumEntropyModelParameters model, DataViewSchema trainSchema) =>
+            new MulticlassPredictionTransformer<MaximumEntropyModelParameters>(Host, model, trainSchema, FeatureColumn.Name, LabelColumn.Name);
+    }
+
+    /// <summary>
+    /// The <see cref="IEstimator{TTransformer}"/> for training a multiclass linear model using the stochastic dual coordinate ascent method.
+    /// The trained model <see cref="LinearMulticlassModelParameters"/> does not produces probabilities of classes, but we can still make decisions
+    /// by choosing the class associated with the largest score.
+    /// </summary>
+    /// <include file='doc.xml' path='doc/members/member[@name="SDCA_remarks"]/*' />
+    public sealed class SdcaNonCalibratedMulticlassClassificationTrainer : SdcaMulticlassClassificationTrainerBase<LinearMulticlassModelParameters>
+    {
+        public class Options : CommonOptions
```

Comment:
>public class Options [](start = 8, length = 20)

sealed?

Reply:
Sure. Thanks.

---
In reply to: [266991181](https://github.com/dotnet/machinelearning/pull/2976#discussion_r266991181) [](ancestors = 266991181)